src/WSRequestHandler_General.cpp

```diff
@@ -345,3 +345,69 @@ RpcResponse WSRequestHandler::OpenProjector(const RpcRequest& request) {
 	obs_frontend_open_projector(type, monitor, geometry, name);
 	return request.success();
 }
+
+/**
+* Executes hotkey routine, identified by hotkey unique name
+*
+* @param {String} `name` Unique name of the hotkey, as defined when registering the hotkey (e.g. "ReplayBuffer.Save")
+*
+* @api requests
+* @name ProcessHotkeyByName
+* @category general
+* @since unreleased
+*/
+RpcResponse WSRequestHandler::ProcessHotkeyByName(const RpcRequest& request) {
+	const char* name = obs_data_get_string(request.parameters(), "name");
+
+	obs_hotkey_t* hk = Utils::FindHotkeyByName(name);
+	if (!hk) {
+		return request.failed("Hotkey not found");
+	}
+	obs_hotkey_trigger_routed_callback(obs_hotkey_get_id(hk), true);
+	return request.success();
+}
+
+/**
+* Executes hotkey routine, identified by bound combination of keys. A single key combination might trigger multiple hotkey routines depending on user settings 
+*
+* @param {String} `key` Main key identifier (e.g. `OBS_KEY_A` for key "A"). Available identifiers [here](https://github.com/obsproject/obs-studio/blob/master/libobs/obs-hotkeys.h)
+* @param {Object (Optional)} `modifiers` Optional key modifiers object: `{"shift":true/false, "alt":true/false, "control":true/false, "command":true/false}`. False entries can be ommitted
+*
+* @api requests
+* @name ProcessHotkeyByCombination
+* @category general
+* @since unreleased
+*/
+RpcResponse WSRequestHandler::ProcessHotkeyByCombination(const RpcRequest& request) {
+	if (!request.hasField("key")) {
+		return request.failed("Missing request key parameter");
+	}
+
+	OBSDataAutoRelease data = obs_data_get_obj(request.parameters(), "modifiers");
+
+	obs_key_combination_t combo = {0};
+	uint32_t modifiers = 0;
+	if (obs_data_get_bool(data, "shift"))
+		modifiers |= INTERACT_SHIFT_KEY;
+	if (obs_data_get_bool(data, "control"))
+		modifiers |= INTERACT_CONTROL_KEY;
+	if (obs_data_get_bool(data, "alt"))
+		modifiers |= INTERACT_ALT_KEY;
+	if (obs_data_get_bool(data, "command"))
+		modifiers |= INTERACT_COMMAND_KEY;
+
+	combo.modifiers = modifiers;
+	combo.key = obs_key_from_name(obs_data_get_string(request.parameters(), "key"));
+
+	if (!modifiers
+		&& (combo.key == OBS_KEY_NONE || combo.key >= OBS_KEY_LAST_VALUE)) {
+		return request.failed("Invalid key-modifier combination");
+	}
+
+	// Inject hotkey press-release sequence
+	obs_hotkey_inject_event(combo, false);
```

Comment:
Just curious, what's the purpose behind setting it `false` first?

Reply:
It's a good question and I expected it because it's definitely not intuitive. Explanation relates to how OBS `handle_binding()` function works, especially this [line](https://github.com/obsproject/obs-studio/blob/b9ea45e4982f4ec26b3f115f0e625289a8f26ae4/libobs/obs-hotkey.c#L1255). Apparently, when using modifiers in the combination, you need to do 2 passes of the function to properly set the` binding->modifiers_match` parameter (which is by default false, and is only potentially true after second pass). So that first call to the `inject_event()` helps initializing the param if needed.

To be honest, I had to dive deep into the OBS code and debug this request thoroughly because I wasn't expecting this line to be needed either, but it's the solution I found to make it work with modifiers.

Reply:
Very interesting. Good work